cli/tools/upgrade.rs

```diff
@@ -120,6 +123,10 @@ impl RealVersionProvider {
 
 #[async_trait(?Send)]
 impl VersionProvider for RealVersionProvider {
+  fn is_release_candidate(&self) -> bool {
+    version::is_release_candidate()
+  }
```

Comment:
Most likely remove; since canary and rc are gonna be pretty much the same and the only difference is that RC is just a designated canary release we might want to skip this particular check. Alternatively this could be an async fn and check against the designated canary to figure out if it's RC.